### PR TITLE
Signin rhp

### PR DESCRIPTION
**Pull request by daordonez11**
### Details
Implementation to send sign in to RHP

### Fixed Issues

$ https://github.com/Expensify/App/issues/20558
PROPOSAL: https://github.com/Expensify/App/issues/20558#issuecomment-1593859489

### Tests

1. Open a public room as unauthenticated user (i.e. https://staging.new.expensify.com/r/8813851443373010)
2. Click on sign-in
3. Click on the back button, should see the public room instead of closing the app
4. Enter sign-in in RHP
//Validate claim

5. Enter a new email and enters the validation code
6. Verify the app loads with the authenticated data
7. Open in an incognito mode, sign in with another account and post something in the public room
8. Verify the message appears in the public room for the first user that claimed the anonymous account
9. Repeat the process again from step 4 to 8, but in step 5 enter the email of an existing account

### Offline tests
N/A anonymous users only work online

### QA Steps
Same as test steps
### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/13544910/ede9af0d-1d6f-4b97-9e3a-daaee155c7d8

[Web Refresh Testing - 8_9_2023, 8_33_09 PM.webm](https://github.com/Expensify/App/assets/13544910/af54a124-1570-44ba-9a92-a9ffdf2c4d4c)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/13544910/3ca3964a-10b6-44fe-b123-ee49bac257ad

[mWeb Refresh - 8_9_2023, 8_38_46 PM.webm](https://github.com/Expensify/App/assets/13544910/6cd993cb-9ebe-4d02-9d34-3861804ba5b1)

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/13544910/4b30a6ee-23f5-4123-b641-da4fd6e8d517

</details>

<details>
<summary>Desktop</summary>

N/A Public rooms are not supported in desktop

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/13544910/8267612c-1f04-4036-8c47-486905228026

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/13544910/a4473f53-9e97-435f-9de1-aaebcd24dc94

</details>

### CODE REVIEW

File: src/pages/signin/SignInPage.js [hunk 1]
```diff
@@ -77,9 +80,10 @@ function getRenderOptions({hasLogin, hasValidateCode, hasAccount, isPrimaryLogin
     };
 }
 
-function SignInPage({credentials, account}) {
+function SignInPage({credentials, account, isInModal}) {
     const {translate, formatPhoneNumber} = useLocalize();
     const {isSmallScreenWidth} = useWindowDimensions();
+    const showSmallScreen = isSmallScreenWidth || isInModal;
```
Comment (mollfpr) [inline src/pages/signin/SignInPage.js:86]: NAB: We should use the `should` or `is` prefix for the boolean value.

File: src/ROUTES.js [hunk 2]
```diff
@@ -187,4 +187,5 @@ export default {
             isSubReportPageRoute: pathSegments.length > 2,
         };
     },
+    SIGNINMODAL: 'signinmodal',
```
Comment (mollfpr) [inline src/ROUTES.js:190]: I think it's better to separate the words with the dash and underscore for the key.

```suggestion
    SIGN_IN_MODAL: 'sign-in-modal',
```

File: src/pages/signin/SignInModal.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import styles from '../../styles/styles';
+
+const propTypes = {};
+
+const defaultProps = {};
+
+function SignInModal() {
+    return (
+        <ScreenWrapper
+            style={[styles.signInModal]}
```
Comment (mollfpr) [inline src/pages/signin/SignInModal.js:15]: We don't need to create a new style for this. We can reuse `styles.highlightBG`.

**Review by mollfpr — CHANGES_REQUESTED** [review]
https://github.com/Expensify/App/assets/25520267/38154462-b057-48f5-a013-22234fadc5ff

@daordonez11 Could you confirm if this issue already notice before and will be fixed later?

**Comment by marcochavezf** [comment]
@daordonez11 @mollfpr for the test steps, also add the case to verify that the logged in user can send a message and other users receive it without refreshing the browser (you can use the same steps from [here](https://github.com/Expensify/App/pull/23120#issue-1810518507)).

**Comment by daordonez11** [comment]
> Screen.Recording.2023-08-07.at.23.16.25.mov 
> @daordonez11 Could you confirm if this issue already notice before and will be fixed later?

@marcochavezf  what do you think of that issue, I have 2 ideas.
1. Send to home if user is already logged in (dismissModal)
2. Use something like authenticated views so Hmmmmm not found is shown.

I think 1 is easier but users would still see for a fraction the view. Which one do you think we should use?

Thanks for the report @mollfpr I did see it once I think it can also be reproduced by pressing back button.

File: src/ROUTES.js (hunk 2 above)
Comment (daordonez11) [reply src/ROUTES.js]: Updated name and references!

File: src/pages/signin/SignInModal.js (hunk 3 above)
Comment (daordonez11) [reply src/pages/signin/SignInModal.js]: Thanks for the idea! Didn't saw it while coding

File: src/pages/signin/SignInPage.js (hunk 1 above)
Comment (daordonez11) [reply src/pages/signin/SignInPage.js]: updated to shouldShowSmallScreen

**Comment by marcochavezf** [comment]
> > Screen.Recording.2023-08-07.at.23.16.25.mov
> > @daordonez11 Could you confirm if this issue already notice before and will be fixed later?
> 
> @marcochavezf what do you think of that issue, I have 2 ideas.
> 
> 1. Send to home if user is already logged in (dismissModal)
> 2. Use something like authenticated views so Hmmmmm not found is shown.
> 
> I think 1 is easier but users would still see for a fraction the view. Which one do you think we should use?
> 
> Thanks for the report @mollfpr I did see it once I think it can also be reproduced by pressing back button.

My first question is, why the validation code is set after refreshing the page? I think it's fine to show the sign-in modal when the user enters the URL, but it's weird that the form is displaying the validation code and trying to submit it

**Comment by daordonez11** [comment]
So basically this occurs because the state of the component is mantained, since this was originally an unprotected view the user does not see it but its just the "final" state of the page. We could use something like `clearLocalSignInData` before leaving the page but tbh I don't think showing the signin should be enabled when you are already logged in. Me recommendation would be to auto dismiss the modal if the view is accessed via url and the user is already logged in.

cc @mollfpr @marcochavezf

**Comment by mollfpr** [comment]
> Me recommendation would be to auto dismiss the modal if the view is accessed via url and the user is already logged in.

@daordonez11 @marcochavezf I agree with this.

**Comment by marcochavezf** [comment]
> > Me recommendation would be to auto dismiss the modal if the view is accessed via url and the user is already logged in.
> 
> @daordonez11 @marcochavezf I agree with this.

Sounds good, let's go with that solution!

**Comment by daordonez11** [comment]
Ok guys had to implement a last detail in order for notifications and new scenarios to work. Final commit includes that implementation. Currently uploading videso with new implementation. So ready again for review with new scenario @mollfpr!

cc @marcochavezf

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by mollfpr — COMMENTED** [review]
Sorry for the delay, I am just feeling better now 🙏

The code mostly looks good, just couples of NABs.

There's a strange bug that showing the `SignInModal` page after sign in.

Steps:
1. Open a public room with anonymous account
2. Clear the browser storage
3. Open the sign-in page and complete the sign-in flow
4. The user will be redirected to the sign-in-modal with the validate code filled.

I believe this is an edge-case, but I'm afraid we will have similar issue with no step reproduce it yet.

https://github.com/Expensify/App/assets/25520267/77b1027e-6eb3-4ed1-b7ba-7018b1807c54

**Comment by daordonez11** [comment]
Hey @mollfpr  thanks for the update and hope you get better! I'll check the NAB's and the error tomorrow, I definitely need to make sure rhp is not shown if you are already logged in, probably with more than one check. I'll let you guys know when it's ready.

cc @marcochavezf

**Comment by daordonez11** [comment]
Thanks for the report @mollfpr the issue was happening because navigation was not ready at that point so dismissModal did nothing. Just uploaded a commit with the NABs and the fix with the navigation promise. The easy way of replicating was just opening a new tab with the sign in modal url 😂.

cc @marcochavezf

**Comment by daordonez11** [comment]
PS: Solved conflict that appeared today!
cc @marcochavezf @mollfpr

**Comment by mollfpr** [comment]
Thanks, @daordonez11. Seems looks good now. I'll be finishing the records in an hour.

**Review by mollfpr — APPROVED** [review]
LGTM and tests well 🚀

**Review by marcochavezf — APPROVED** [review]
Looks great, thank you guys, you're awesome!

**Comment by roryabraham** [comment]
Hi 👋🏼 

The SignInPageLayout was dependent upon `isSmallScreenWidth` for it's layout to look correct. However, this displayed the SignInPageLayout in the RHP. The intention was for it to look and behave how it would in a small screen, but the `isSmallScreenWidth` was still true on wide screens.

We addressed this bug in https://github.com/Expensify/App/issues/28025

**Comment by shubham1206agra** [comment]
Hello

You have forgot to take out extra padding for RHP, which caused https://github.com/Expensify/App/issues/31445.

File: src/libs/Navigation/AppNavigator/AuthScreens.js [hunk 4]
```diff
@@ -48,6 +48,11 @@ Onyx.connect({
         }
 
         currentAccountID = val.accountID;
+        if (Navigation.isActiveRoute(ROUTES.SIGN_IN_MODAL)) {
+            // This means sign in in RHP was successful, so we can dismiss the modal and subscribe to user events
+            Navigation.dismissModal();
```
Comment (Pujan92) [inline src/libs/Navigation/AppNavigator/AuthScreens.js:53]: Seems we are closing the modal early which causes the issue https://github.com/Expensify/App/issues/36976

**Pull request merged**